src/main/java/library/app/spring/controller/BookController.java

```diff
@@ -0,0 +1,49 @@
+package library.app.spring.controller;
+
+import java.util.List;
+
+import library.app.spring.entity.Book;
+import library.app.spring.service.BookService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@Controller
+@RequestMapping("/book")
+public class BookController {
+
+    @Autowired
+    private BookService bookService;
+
+    @GetMapping("/allBooks")
```

Comment:
"/all" is enough. Full path would be "/book/all". Method is in the Book controller so no need to specify "/allBooks"

Reply:
complete